tools/python-integration-tests/test.py

```diff
@@ -0,0 +1,378 @@
+        time.sleep(3)
+
+        # Create an SSH client
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        # Load the private key
+        key = paramiko.RSAKey.from_private_key_file(key_path)
+
+        # Connect to the VM 
+        ssh.connect("localhost", port=10022, username=self.username, pkey=key)
+    
+        return ssh
+        
+    def close_tunnel(self):
+        if self.tunnel:
+            self.tunnel.terminate()
+            self.tunnel.wait()
+            self.tunnel = None
+
+    def destroy(self):
+        cmd = [
+              "./gcluster",
+              "destroy",
+              self.deployment_name,
+              "--auto-approve"
+          ]
+
+        subprocess.run(cmd, check=True, cwd=self.workspace)
+        os.remove(f"{self.deployment_name}.tgz")
+        shutil.rmtree(self.deployment_name)
+
+
+class Test:
+    def __init__(self, ssh, deployment):
+        self.ssh = ssh
+        self.deployment = deployment
+        self.job_list = {}
+
+    def get_directory_contents(self):
+        stdin, stdout, stderr = self.ssh.exec_command('ls -l')
+        return stdout.read().decode()
+
+    def get_sinfo(self):
+        stdin, stdout, stderr = self.ssh.exec_command('sinfo')
+        return stdout.read().decode()
+
+    def print_jobs(self):
+        for job in self.job_list.items():
+            print(job)
+        print()
+
+    def print_job_output(self, job_id: str):
+        stdin, stdout, stderr = self.ssh.exec_command(f"cat slurm-{job_id}.out")
+        print(f"slurm-{job_id}.out:", stdout.read().decode())
+
+    def get_slurm_topology(self):
+        stdin, stdout, stderr = self.ssh.exec_command("scontrol show topo")
+        return stdout.read().decode() 
+
+    def monitor_squeue(self):
+        # Monitors squeue and updates self.job_list until all running jobs are complete.
+        lines = []
+
+        while True:
+            stdin, stdout, stderr = self.ssh.exec_command('squeue')
+
+            lines = stdout.read().decode().splitlines()[1:] # Skip header
+
+            if not lines:
+                break
+            for line in lines:
+                parts = line.split()
+                job_id, partition, _, _, state, times, nodes, nodelist = line.split()
+
+                if job_id not in self.job_list:
+                    print(f"Job id {job_id} is not recognized.")
+                else:
+                    self.job_list[job_id].update({
+                        "partition": partition,
+                        "state": state,
+                        "time": times,
+                        "nodes": nodes,
+                        "nodelist": nodelist,
+                    })
+            time.sleep(5)
+
+    def is_job_complete(self, job_id: str):
+        # Checks if a job successfully completed.
+        stdin, stdout, stderr = self.ssh.exec_command(f'scontrol show job {job_id}')
```

Comment:
You can use `scontrol bla-bla --json` to avoid regexing.

Reply:
Great suggestion, changed to used json.